cumulus_etl/fhir/fhir_client.py

```diff
@@ -86,7 +90,15 @@ async def __aexit__(self, exc_type, exc_value, traceback):
             await self._session.aclose()
 
     async def request(
```

Comment:
Let me lightly explain some of the refactoring here.

Before, FhirClient.request was a single request. The retry logic was in the bulk export code, which retried on both error cases and the expected `202` status code for "still working on it".

Now, I've separated the bulk export retry logic into error-path and 202-path, and moved the error-path into this FhirClient.request method. So now any code making requests to the server can enable retries if they pass in a `retry_delays` array of retry timeouts.